lib/RestructureCFG/InlineDivergentScopesPass.cpp

```diff
@@ -279,6 +279,49 @@ electDivergence(BasicBlock *Candidate,
   return std::nullopt;
 }
 
+/// Helper function which create a single entry point block where a divergence
+/// is entered into by multiple successors of a `Conditional`
+static void createHead(BasicBlock *Conditional,
+                       DivergenceDescriptor &Divergence,
+                       BasicBlock *PlaceHolderTarget) {
+  LLVMContext &Context = getContext(Conditional);
+  Function *F = Conditional->getParent();
+  Instruction *ConditionalTerminator = Conditional->getTerminator();
+  BasicBlock *Head = BasicBlock::Create(Context,
+                                        Conditional->getName() + "_head_ids",
+                                        F);
+  Instruction *HeadTerminator = ConditionalTerminator->clone();
+  IRBuilder<> HeadBuilder(Head);
+  HeadBuilder.Insert(HeadTerminator);
+
+  // Collect the `Successor`s composing the local `Divergence`
+  SmallSet<BasicBlock *, 4> LocalDivergentSuccessors;
+  LocalDivergentSuccessors.insert(Divergence.DivergentSuccessors.begin(),
+                                  Divergence.DivergentSuccessors.end());
```

Comment:
Doesn't copy-assignment work?
```c++
SmallSet<BasicBlock *, 4> LocalDivergentSuccessors = Divergence.DivergentSuccessors;
```

Reply:
Yeah, you are right, changed this here https://github.com/revng/revng/blob/3a9379ef740d1d2b9f7bcad02cc962719251e02a/lib/RestructureCFG/InlineDivergentScopesPass.cpp#L319